pkg/innerring/deploy.go

```diff
@@ -64,12 +64,20 @@ func (x *fsChain) SubscribeToNewBlocks() (<-chan *block.Block, error) {
 		return ch, nil
 	}
 
-	err := x.client.ReceiveBlocks()
+	err := x.client.ReceiveHeaders()
 	if err != nil {
 		return nil, fmt.Errorf("listen to new blocks over Neo RPC WebSocket multi-endpoint: %w", err)
 	}
 
-	_, ch, _ := x.client.Notifications()
+	_, headerCh, _ := x.client.Notifications()
+	ch := make(chan *block.Block)
+	go func() {
+		// Client closure will stop it
+		for h := range headerCh {
```

Comment:
i'd `close(ch)` after the loop to notify the callers

Reply:
Makes sense, done. Obviously this code is temporary (https://github.com/nspcc-dev/neofs-contract/pull/471).